blog/2021/apisix.adoc

```diff
@@ -0,0 +1,184 @@
+
+image::${blogImages}/apisix/create-realm.png[Create realm]
+
+image::${blogImages}/apisix/edit-realm-title.png[Edit realm title]
+
+
+===== Create a Client
+The next step is to create the `OpenID Connect Client`. In Keycloak, Client means a client that is allowed to initiate authentication to Keycloak.
+
+In this example scenario, `Apache APISIX` is equivalent to a client that is responsible for initiating authentication requests to Keycloak, so we create a Client with the name `apisix`. More details about the Client can be found in https://www.keycloak.org/docs/latest/server_admin/#_oidc_clients[Keycloak OIDC Clients].
+
+image::${blogImages}/apisix/create-openid-client.png[Create OpenID Client]
+
+image::${blogImages}/apisix/create-client-title.png[Create Client title]
+
+===== Configure the client
+After the Client is created, you need to configure the Apache APISIX access type for the Client.
+
+In Keycloak, there are three types of Access Type:
+
+. *Confidential*: which is used for applications that need to perform browser login, and the client will get the `access token` through `client secret`, mostly used in web systems rendered by the server.
+. *Public*: for applications that need to perform browser login, mostly used in front-end projects implemented using vue and react.
+. *Bearer-only*: for applications that don't need to perform browser login, only allow access with `bearer token`, mostly used in RESTful API scenarios.
+
+For more details about Client settings, please refer to https://www.keycloak.org/docs/latest/server_admin/#advanced-settings[Keycloak OIDC Clients Advanced Settings].
+
+Since we are using Apache APISIX as the Client on the server side, we can choose either "confidential" Access Type or "Bearer-only" Access Type. For the demonstration below, we are using "confidential" Access Type as an example.
+
+
+image::${blogImages}/apisix/set-client-type.png[Set Client type]
+
+===== Create Users
+Keycloak supports interfacing with other third-party user systems, such as Google and Facebook, or importing or manually creating users using LDAP . Here we will use "manually creating users" to demonstrate.
+
+image::${blogImages}/apisix/create-user.png[Create user]
+
+image::${blogImages}/apisix/add-user-info.png[Add user info]
+
+Then set the user's password in the Credentials page.
+
+image::${blogImages}/apisix/set-user-password.png[Set user password]
+
+==== Create Routes
+After Keycloak is configured, you need to create a route and open the `Openid-Connect` plugin . For details on the configuration of this plugin, please refer to the https://apisix.apache.org/docs/apisix/plugins/openid-connect[Apache APISIX OpenID-Connect plugin].
+
+===== Get client_id and client_secret
+![Get Client infos](https://static.apiseven.com/202108/1639101454807-ff8c8b77-bdac-4ac6-966e-a2f5e2418b7a.png)
```

Comment:
This doesn't render, and due to CSP headers won't be permitted either. So can you add this image to the PR as well?

Reply:
Yes this image was already added in line 116. This line 114 should have been removed.